### PR TITLE
feat: adding dns01 solver for cert manager

### PR DESCRIPTION
**Pull request by gwenall**
# Pull request title

## Description

Please explain the changes you made here and link to any relevant issues.

### Checklist

- [ ] CI tests are passing
- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/particuleio/terraform-kubernetes-addons/#doc-generation

### CODE REVIEW

File: modules/scaleway/cert-manager.tf [hunk 1]
```diff
@@ -15,6 +15,13 @@ locals {
       acme_http01_enabled       = false
       acme_http01_ingress_class = "nginx"
       acme_dns01_enabled        = false
+      acme_dns01_provider       = ""
+      acme_dns01_hosted_zone_id = ""
+      acme_dns01_aws_secret     = ""
+      acme_dns01_aws_access_key_id  = ""
+      acme_dns01_aws_access_key_secret = ""
+      acme_dns01_region         = ""
+      cluster_issuer_assume_role_arn = ""
```
Comment (dud225) [inline modules/scaleway/cert-manager.tf:24]: not used

Comment (dud225) [inline modules/scaleway/cert-manager.tf:24]: not used

Comment (dud225) [inline modules/scaleway/cert-manager.tf:24]: `cluster_issuer_assume_role_arn` is not used anywhere

Comment (gwenall) [reply modules/scaleway/cert-manager.tf]: removed, thanks

File: modules/scaleway/cert-manager.tf [hunk 2]
```diff
@@ -15,6 +15,12 @@ locals {
       acme_http01_enabled       = false
       acme_http01_ingress_class = "nginx"
       acme_dns01_enabled        = false
+      acme_dns01_provider       = ""
+      acme_dns01_hosted_zone_id = ""
+      acme_dns01_aws_secret     = ""
```
Comment (ArchiFleKs) [inline modules/scaleway/cert-manager.tf:20]: I find this not explicit, could we use `acme_dns01_aws_secret_name`

File: modules/scaleway/cert-manager.tf [hunk 3]
```diff
@@ -15,6 +15,12 @@ locals {
...
+      acme_dns01_aws_access_key_id  = ""
```
Comment (ArchiFleKs) [inline modules/scaleway/cert-manager.tf:21]: same here `acme_dns01_aws_access_key_id_secret_key`

File: modules/scaleway/cert-manager.tf [hunk 4]
```diff
@@ -15,6 +15,12 @@ locals {
...
+      acme_dns01_aws_access_key_secret = ""
```
Comment (ArchiFleKs) [inline modules/scaleway/cert-manager.tf:22]: `acme_dns01_aws_secret_access_key_secret_key`

File: modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl [hunk 5]
```diff
@@ -46,6 +61,28 @@ spec:
...
+            name: ${acme_dns01_aws_secret}
+            key: ${acme_dns01_aws_access_key_id}
+          secretAccessKeySecretRef:
+            name: ${acme_dns01_aws_secret}
+            key: ${acme_dns01_aws_access_key_secret}
+    %{ else }
+    %{if acme_dns01_provider == "google" }
+    - dns01:
+        clouddns:
+          project: '${acme_dns01_google_project}'
```
Comment (ArchiFleKs) [inline modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl:81]: Is this undefined?

File: modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl [hunk 6]
```diff
@@ -46,6 +61,28 @@ spec:
...
+          serviceAccountSecretRef:
+            name: '${acme_dns01_google_secret}'
+            key: '${acme_dns01_google_service_account_key}'
```
Comment (ArchiFleKs) [inline modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl:84]: Is this undefined?

File: modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl [hunk 7]
```diff
@@ -46,6 +61,28 @@ spec:
...
+          secretAccessKeySecretRef:
+            name: ${acme_dns01_aws_secret}
+            key: ${acme_dns01_aws_access_key_secret}
+    %{ else }
+    %{if acme_dns01_provider == "google" }
+    - dns01:
+        clouddns:
+          project: '${acme_dns01_google_project}'
+          serviceAccountSecretRef:
+            name: '${acme_dns01_google_secret}'
```
Comment (ArchiFleKs) [inline modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl:83]: Is this undefined?

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

File: modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl (hunk 6 above)
Comment (gwenall) [reply modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl:84]: added local in cert-manager.tf

File: modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl (hunk 5 above)
Comment (gwenall) [reply modules/scaleway/templates/cert-manager-cluster-issuers.yaml.tpl:81]: added local in cert-manager.tf

**Review by ArchiFleKs — APPROVED** [review]

**Pull request merged**